tests/common/common_vec.h

```diff
@@ -441,7 +441,8 @@ asType check_as_result(sycl::vec<vecType, N> inputVec,
   for (size_t i = 0; i < N; ++i) {
     tmp_ptr[i] = getElement(inputVec, i);
   }
-  asType* exp_ptr = reinterpret_cast<asType*>(tmp_ptr);
+  asType exp_ptr[asN];
+  memcpy(exp_ptr, tmp_ptr, sizeof(exp_ptr));
```

Comment:
Can we use C++ std::variant here instead?

Reply:
Does std::variant allow this kind of type punning? I thought it was designed to make it impossible to access the value using the wrong type like this.

Reply:
You are right. Sorry for a senseless question.